docs/tutorial/common_hpc_dependencies.rst

```diff
@@ -0,0 +1,173 @@
+    ``BLT_EXPORT_THIRDPARTY`` option to ``ON``.  As with other
+    EXPORTABLE targets created by ``blt_import_library()``, these targets should be prefixed with the name of
+    the project.  Either the ``EXPORT_NAME`` target property or the ``NAMESPACE`` option to CMake's ``install``
+    command can be used to modify the name of an installed target.  See  :ref:`ExportingTargets:`
+    for more info.
+
+You have already seen one use of ``DEPENDS_ON`` for a BLT dependency, ``gtest``, in ``test_1``:
+
+.. literalinclude:: calc_pi/CMakeLists.txt 
+   :start-after: _blt_tutorial_calcpi_test1_executable_start
+   :end-before:  _blt_tutorial_calcpi_test1_executable_end
+   :language: cmake
+
+
+``gtest`` is the name for the GoogleTest target in BLT registered via 
+``blt_register_library()``. Even though GoogleTest is built-in and uses CMake,
+``blt_register_library()`` allows us to easily set defines needed by all dependent
+targets.
+
+
+MPI
+~~~
+
+Our next example, ``test_2``, builds and tests the ``calc_pi_mpi`` library,
+which uses MPI to parallelize the calculation over the integration intervals.
+
+
+To enable MPI, we set ``ENABLE_MPI``, ``MPI_C_COMPILER``, and ``MPI_CXX_COMPILER``
+in our host config file. Here is a snippet with these settings for LLNL's Pascal Cluster:
+
+.. literalinclude:: ../../host-configs/llnl/toss_3_x86_64_ib/gcc@4.9.3_nvcc.cmake
+   :start-after: _blt_tutorial_mpi_config_start
+   :end-before:  _blt_tutorial_mpi_config_end
+   :language: cmake
+
+
+Here, you can see how ``calc_pi_mpi`` and ``test_2`` use ``DEPENDS_ON``:
+
+.. literalinclude:: calc_pi/CMakeLists.txt 
+   :start-after: _blt_tutorial_calcpi_test2_executable_start
+   :end-before:  _blt_tutorial_calcpi_test2_executable_end
+   :language: cmake
+
+
+For MPI unit tests, you also need to specify the number of MPI Tasks
+to launch. We use the ``NUM_MPI_TASKS`` argument to ``blt_add_test()`` macro.
+
+.. literalinclude:: calc_pi/CMakeLists.txt 
+   :start-after: _blt_tutorial_calcpi_test2_test_start
+   :end-before:  _blt_tutorial_calcpi_test2_test_end
+   :language: cmake
+
+
+As mentioned in :ref:`AddingTests`, GoogleTest provides a default ``main()``
+driver that will execute all unit tests defined in the source. To test MPI code,
+we need to create a main that initializes and finalizes MPI in addition to Google
+Test. ``test_2.cpp`` provides an example driver for MPI with GoogleTest.
+
+.. literalinclude:: calc_pi/test_2.cpp
+   :start-after: _blt_tutorial_calcpi_test2_main_start
+   :end-before:  _blt_tutorial_calcpi_test2_main_end
+   :language: cpp
+
+.. note::
+  While we have tried to ensure that BLT chooses the correct setup information
+  for MPI, there are several niche cases where the default behavior is
+  insufficient. We have provided several available override variables:
+  
+  * ``BLT_MPI_COMPILE_FLAGS``
+  * ``BLT_MPI_INCLUDES``
+  * ``BLT_MPI_LIBRARIES``
+  * ``BLT_MPI_LINK_FLAGS``
+  
```

Comment:
Should we add a note that BLT's MPI variables work around changes in CMake's MPI support over several versions?